sentry-types/src/protocol/envelope.rs

```diff
@@ -1,16 +1,82 @@
-use std::io::Write;
+use std::{io::Write, path::Path};
 
+use serde::Deserialize;
+use thiserror::Error;
 use uuid::Uuid;
 
 use super::v7::{Attachment, Event, SessionAggregates, SessionUpdate, Transaction};
 
+/// Raised if a envelope cannot be parsed from a given input.
+#[derive(Debug, Error)]
+pub enum EnvelopeError {
+    /// Unexpected end of file
+    #[error("unexpected end of file")]
+    UnexpectedEof,
+    /// Missing envelope header
+    #[error("missing envelope header")]
+    MissingHeader,
+    /// Missing item header
+    #[error("missing item header")]
+    MissingItemHeader,
+    /// Missing newline after header or payload
+    #[error("missing newline after header or payload")]
+    MissingNewline,
+    /// Invalid envelope header
+    #[error("invalid envelope header")]
+    InvalidHeader(#[source] serde_json::Error),
+    /// Invalid item header
+    #[error("invalid item header")]
+    InvalidItemHeader(#[source] serde_json::Error),
+    /// Invalid item payload
+    #[error("invalid item payload")]
+    InvalidItemPayload(#[source] serde_json::Error),
+}
+
+#[derive(Deserialize)]
+struct EnvelopeHeader {
```

Comment:
Should we care about more optional attributes like `sent_at`, `dsn`, etc? Probably not now, since envelope "round-trips" are not really a goal for now.

Reply:
We would need to change `to_writer` as well to handle this, as we currently ignore anything else other than `event_id`.